src/main/php/unittest/TestSuite.class.php

```diff
@@ -273,9 +272,8 @@ protected function runInternal($test, $result) {
     );
 
     $timer= new Timer();
-    $report= function($type, $outcome, $arg) use($result, $timer, &$t) {
-      $timer->stop();
-      $this->notifyListeners($type, [$result->set($t, new $outcome($t, $arg, $timer->elapsedTime()))]);
+    $record= function($type, $outcome) use($result, &$t) {
+      $this->notifyListeners($type, [$result->set($t, $outcome)]);
       \xp::gc();
```

Comment:
This could even further be simplified if TestOutcome defined a test() accessor to return the TestCase: `$result->set($outcome->test(), $outcome)` => `$result->record($outcome)`.